playground/doc/playground.mdx

```diff
@@ -3,36 +3,66 @@ import { mdx, MDXCreateElement } from '@mdx-js/react';
 import { Props, Description } from '~/props/dist/index';
 import { Playground } from '../dist';
 import { Button } from '../test/index';
+import styles from './styles.css';
 import { CssLayout } from '~/mdx-layout-css';
 export default CssLayout;
 
 # Playground Component
 
 <Description of={Playground} />
 
-## Usage
+## Usage with Shadow DOM
+
+```tsx
+import { Playground } from '@divriots/dockit-react';
+import styles from './styles.css';
+
+<Playground
+  styles={styles}
+  code={`
+<button className="btn">Button</button>
+<button className="btn btn-primary">Primary</button>
+<button className="btn btn-danger">Danger</button>
+  `}
+/>;
+```
+
+<Playground
+  styles={styles}
+  code={`
+<button className="btn">Button</button>
+<button className="btn btn-primary">Primary</button>
+<button className="btn btn-danger">Danger</button>
+  `}
+/>
+
+## Usage with global styles
+
+This includes the usage with most CSS-in-JS libraries which rely on global style tag.
 
 ```tsx
 import { Playground } from '@divriots/dockit-react';
 import { Button } from './Button';
 
 <Playground
+  disableStylesEncapsulation
   scope={{ Button }}
   code={`
-  <Button>Button</Button>
-  <Button primary>Primary</Button>
-  <Button danger>Danger</Button>
+<Button>Button</Button>
+<Button primary>Primary</Button>
+<Button danger>Danger</Button>
```

Comment:
I was doubting between making encapsulation a default and after @georges-gomes words on that I decided to give it a god and bump the version to `0.1.0` (considerer major bump for 0.x.x versions, so should not affect AOS)

I decided to use `disableStylesEncapsulation` (vs `disableShadowDOM`), since indeed iframes can be added later as an option, or even by default, if Shadow DOM won't solve all issues for our users, e.g. because as we know the inherited styles do pierce Shadow DOM too

but I also decided to give more thinking to CSS-in-JS support which I think we should be careful with too

most CSS-in-JS libs create style tag or multiple tags in page's `<head>`
the obvious exceptions are the ones which use inline styles via `style` attribute, e.g. `radium`
they hate/ignore ShadowDOM
=> then by defaulting to encapsulation we will force users of libs like `emotion` or `styled-components` to use `disableStylesEncapsulation` which might be confusing for them, especially given `<Playground>` is JSX!

so I started thinking of having 2 JSX components: leave `<Playground>` as it works in master right now and introduce the new behavior with `<EncapsulatedPlayground>` (name is up for debate)

Reply:
wdyt @gqio @georges-gomes @muryoh  @gluck?

Reply:
Why not using just the styles props to determine if someone wants to use or not the encapsulation ?

Having 2 different is also a good idea since you can enforce the property `styles`
But I think a cooler would be better :)

ShadowPlayground/ScopedPlayground/PlaygroundStyles/PlaygroundBlock/CustomPlayground ...

Reply:
Something is troubling me now with the root of the issue.

In both Tailwind and System-ui scenario, you want to have a everything on the page without encapsulation. Because all the styles are managed globally (for tailwind) and inside the Theme provider (for system-ui).

So the encapsulation is not required and even not recommended in order to have the components really in situe.

So I think our current approach with Playground is correct. The fix is probably somewhere else.

What is the exact problem AOS is facing? Do we have a link to example?

Reply:
> Why not using just the styles props to determine if someone wants to use or not the encapsulation ?

less magic and actually you might not want `styles` and yet want encapsulation, our example in docs made me see it

Reply:
> ShadowPlayground

only if for iframe we decide to go for `IframePlayground`
which is fine by me, because the alternative would be `<EncapsulatedPlayground mode="iframe">` or `<EncapsulatedPlayground mode="shadow">` and decision about defaults again

> ScopedPlayground

confusing, because we use a term `scope` for passing components

> PlaygroundStyles/PlaygroundBlock/CustomPlayground ...

no, sorry 😛

Reply:
I'm starting to like `<ShadowPlayground>` and a potential addition of `<IframePlayground>` in the future

Reply:
@georges-gomes we had endless discussions about it and could not find anything better

the problem lies in the fact that they want to styles documentation as well and use tailwind itself for it
but additionally they require `tailwind/typography` plugin which adds opinionated styles to most tags, including img which was the original bug report fixed in https://github.com/divriots/dockit-react/pull/79/files

alternative fixes:
- adjust `typography` plugin CSS in such a way that it effects only the needed tags, but doesn't go into `<Playground>` => I tried, but couldn't make it work for various reasons (in simple world this would work `.prose > h1`, but not `.prose > td`... another one I tried was `.prose :not(.playground) h2` and some similar selectors, no luck either due to `:not(.playground)` matching any div in between which will be there for sure
- get rid of `typography` plugin and force the usage of classes on `h1`, `h2`, `p` to style them in the docs => arguably even worse comparing to this PR idea

Reply:
@gqio @georges-gomes I think I overlooked one possibility here: people don't write `<h1>`, `<h2>`, `<p>` by hand, we generate it, so we can generate also `<h1 class="whatever">` and add a postprocessor to `tailwind/typography` which will adapt selectors accordingly

wdyt?

Reply:
actually even if I make it work, I don't think I can use `tailwind/typography` in the same way AOS has it right now in their `tailwind.config.js`, because it breaks the abstraction of MDX and exposes it's custom classes to user-space `tailwind.config.js`

I was thinking of hardcoding a custom style (tailwind based or not) for MDX (and MD) which will remove the need to use `tailwind/typography` in the user-space, but... I still believe `<ShadowPlayground>` is a much better, generic, long-term approach, even for tailwind users

Reply:
There are a few things to consider:

If tailwind/typography is part of the DS then potentially it means that the components could be inside a prose in AOS app as well and have the same problem. In this case, the fact that the story is not wrapped in prose would be the problem.

But if the tailwind/typography is only used for the DS Doc then is shouldn't be part of the tailwind.config.js because it's only docs-related and should be generated separately. In this scenario it makes sense to have prose on the docs and isolate the components that are not supposed to be used with prose.

We are most likely in the second scenario for AOS which let me question the presence of typography plugin in the AOS DS config. But it's convenient for sure...

As crazy at it sounds the real solution would be to have the playground in lightdom because they need the css in head of tailwind and have the content generated by markdown in shadowDOM with exclusive styles of typography (prose).

Not an easy problem but iframe is a more generic solution. ShadowDOM will not work for some frameworks.

Reply:
FYI: AFAIK @gqio added `tailwind/typography` for AOS docs only (it is kinda workaround), so it's not part of their DS, hence such a solution and all considerations in this PR

Reply:
> ShadowDOM will not work for some frameworks.

I suggest to take baby steps. `<ShadowPlayground>` gives nice encapsulation already and covers a lot, iframes can be supported later as soon as we understand where they are needed and what problem we need to solve with them, so we will be able to provide a proper API for it

Reply:
I learned more about MDX features and see that when MD file is imported into the app, you can use your own components for `h1`, `h2`, `p` etc... which makes styling them quire easy without too much custom logic

I wanna implement this as well, because one way or another we need to style docs in a nice way and we should not ask each DS to create their own styles for docs, unless they want to themselves and then it's a whole different story of course which AFAIK wasn't not the case for AOS

Reply:
I'm giving up (at least for now) on the idea to style all elements of the documentation UI via some CSS-in-JS solution to avoid including global styles like typography. The main reason: AOS uses responsiveness, which comes from `tailwind/typography` out-of-the-box, and it looks to be really needed. Together with the need to remake `<NavBar>` and all other elements of `dockit-react/mdx-layout-*` which all heavily rely on global styles, it seems like a task on it's own and I think deserves a proper brainstorm/discussion separately from this thread/PR. I'll make an issue for that.

I see the `<ShadowPlayground>` as a useful tool on its own, which AOS and other users who prefer to style their docs via globals styles will appreciated regardless of how we style our `dockit`. With `<ShadowPlayground>` AOS will be able to do the following:

- remove typography plugin from tailwind config
- include `tailwind/typography` from a CDN as a simple CSS file, e.g.:
  ```ts
  // AOS "mdx-layout/index.ts"
  import 'https://cdn.jsdelivr.net/npm/@tailwindcss/typography@0.4.1/dist/typography.css';
  ```
  This won't use the same tokens from their DS, but that's not needed for docs purposes.
  If they want to have this, they should accept the fact that `tailwind/typography` becomes essential part of their DS and tailwind being a global CSS DS will always make styles leak into their playground. Our job is to explain it and not try to find [some ugly hacks](https://github.com/divriots/dockit-react/pull/79/files).
- use `<ShadowPlayground>` and pass styles from the updated DS which doesn't include `tailwind/typography` anymore

I'll do some renaming as discussed in this thread and let you know when to do final review. If any questions, please discord me and let's have a call, because I think to understand my final plan you need to dive deeper into the problematic which I can help with much faster with a call rather than trying to write books here.